sdk/python/kubeflow/metadata/metadata.py

```diff
@@ -26,38 +27,67 @@
 RUN_PROPERTY_NAME = '__kf_run__'
 ALL_META_PROPERTY_NAME = '__ALL_META__'
 
+class Store(object):
+  """Metadata Store that connects to the Metadata gRPC service."""
+
+  def __init__(self,
+               grpc_host="metadata-grpc-service.kubeflow",
+               grpc_port=8080,
+               root_certificates = None,
+               private_key = None,
+               certificate_chain = None):
+
+    """
+    Args:
+      grpc_host {str} -- Required gRPC service host, e.g.
+                         "metadata-grpc-service.kubeflow".
+      grpc_host {int} -- Required gRPC service port.
+      root_certificates {str} -- Optional SSL certificate for secure connection.
+      private_key {str} -- Optional private_key for secure connection.
+      certificate_chain {str} -- Optional certificate_chain for secure connection.
+    """
+    config = mlpb.MetadataStoreClientConfig()
+    config.host = grpc_host
+    config.port = grpc_port
+    if root_certificates or private_key or certificate_chain:
+      config.ssl_config = config.SSLConfig()
+      config.ssl_config.client_key = private_key
+      config.ssl_config.custom_ca = root_certificates
+      config.ssl_config.server_cert = certificate_chain
+    self.store = metadata_store.MetadataStore(
+      config, disable_upgrade_migration=False)
+
 class Workspace(object):
   """
   Groups a set of runs of pipelines, notebooks and their related artifacts
   and executions.
   """
 
   def __init__(self,
-               backend_url_prefix=None,
+               store= None,
                name=None,
                description=None,
-               labels=None):
+               labels=None,
+               backend_url_prefix=None):
     """
     Args:
-      backend_url_prefix {str} -- Required URL prefix pointing to the metadata
-                                  backend, e.g. "127.0.0.1:8080".
+      backend_url_prefix {str} -- deprecated.
       name {str} -- Required name for the workspace.
       description {str} -- Optional string for description of the workspace.
       labels {object} Optional key/value string pairs to label the workspace.
     """
-    if backend_url_prefix is None or type(backend_url_prefix) != str:
-          raise ValueError("'backend_url_prefix' must be set and in string type.")
+    if backend_url_prefix:
+          raise ValueError("""'backend_url_prefix' is deprecated. Please set
```

Comment:
is that better to WARNING here and remove from next release?

Reply:
My original plan is to do what you suggested, but realized it would require to keep two ways of calling backends and the code would look messy.

Whoever want to use the previous SDK should ping down the version.